modules/minimp3/audio_stream_mp3.cpp

```diff
@@ -144,6 +144,31 @@ void AudioStreamPlaybackMP3::tag_used_streams() {
 	mp3_stream->tag_used(get_playback_position());
 }
 
+_FORCE_INLINE_ int AudioStreamPlaybackMP3::get_frames_per_beat() const {
+	ERR_FAIL_COND_V_MSG(mp3_stream->get_bpm() <= 0, -1, "bpm needs to be assigned in import settings");
+	ERR_FAIL_COND_V_MSG(mp3_stream->sample_rate <= 0, -1, "AudioStream sampling rate is 0");
+	return mp3_stream->sample_rate * 60 / mp3_stream->get_bpm();
+}
+
+int AudioStreamPlaybackMP3::get_current_beat() const {
+	ERR_FAIL_COND_V_MSG(mp3_stream->bar_beats <= 0, -1, "bar_beats needs to be assigned in import settings");
+	return (frames_mixed / get_frames_per_beat()) % mp3_stream->bar_beats;
+}
+
+int AudioStreamPlaybackMP3::get_current_bar() const {
+	ERR_FAIL_COND_V_MSG(mp3_stream->bar_beats <= 0, -1, "bar_beats needs to be assigned in import settings");
+	return (frames_mixed / get_frames_per_beat()) / mp3_stream->bar_beats;
+}
+
+float AudioStreamPlaybackMP3::get_beat_progress() const {
+	return fmod(frames_mixed / (float)get_frames_per_beat(), 1.0);
```

Comment:
```suggestion
	return Math::fmod(frames_mixed / (float)get_frames_per_beat(), 1.0f);
```
We use our own methods for these